File: ios/DoMemory/DoMemory/Services/Levels/EmojiPool.swift
//
//  EmojiPool.swift
//  DoMemory
//
//  Shared emoji pool for procedurally generated boards (Daily Challenge,
//  Levels).
//

/// Curated, visually distinct emoji pool. Order is fixed so seeded
/// selection is reproducible across app versions; only append, never reorder.
enum EmojiPool {
    static let all: [String] = [
        "😀", "😎", "🥳", "😍", "🤓", "😴", "🤖", "👻", "💩", "🐶",
        "🐱", "🦊", "🐻", "🐼", "🐸", "🐵", "🦁", "🐯", "🦄", "🐝",
        "🐢", "🐙", "🦋", "🌵", "🌸", "🍄", "🍎", "🍌", "🍉", "🍓",
        "🍕", "🍔", "🌮", "🍦", "🎈", "⚽️", "🚀", "⭐️", "🌈", "🔥",
        "💎", "🎸", "🎲", "🎯", "👑", "🎁", "❤️", "⚡️"
    ]
}

File: android/app/src/main/java/com/ezequielbrrt/domemory/core/rng/EmojiPool.kt
package com.ezequielbrrt.domemory.core.rng

/**
 * The 48 curated, visually distinct emoji used by every generated board (spec 6.2).
 *
 * **Order is part of the contract** — seeded selection depends on it. Only ever
 * append; never reorder or remove.
 */
object EmojiPool {
    val all: List<String> = listOf(
        "😀", "😎", "🥳", "😍", "🤓", "😴", "🤖", "👻", "💩", "🐶",
        "🐱", "🦊", "🐻", "🐼", "🐸", "🐵", "🦁", "🐯", "🦄", "🐝",
        "🐢", "🐙", "🦋", "🌵", "🌸", "🍄", "🍎", "🍌", "🍉", "🍓",
        "🍕", "🍔", "🌮", "🍦", "🎈", "⚽️", "🚀", "⭐️", "🌈", "🔥",
        "💎", "🎸", "🎲", "🎯", "👑", "🎁", "❤️", "⚡️",
    )

    val size: Int get() = all.size
}
